eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/BomGenerator.java

```diff
@@ -0,0 +1,198 @@
+                                        case "azure-sdk-all":
+                                        case "azure-sdk-parent":
+                                        case "azure-client-sdk-parent":
+                                            break;
+                                        default:
+                                            if (EXCLUSION_LIST.contains(artifactId)
+                                                || artifactId.contains(AZURE_PERF_LIBRARY_IDENTIFIER)
+                                                || (artifactId.contains(AZURE_TEST_LIBRARY_IDENTIFIER) && !artifactId.equalsIgnoreCase(AZURE_CORE_TEST_LIBRARY))) {
+                                                logger.info("Skipping dependency {}:{}", groupId, artifactId);
+                                                continue;
+                                            }
+
+                                            inputDependencies.add(dependency);
+                                            break;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        } catch (IOException exception) {
+            exception.printStackTrace();
+        }
+
+        return inputDependencies;
+    }
+
+    private TreeSet<BomDependency> resolveExternalDependencies() {
+        TreeSet<BomDependency> externalDependencies = new TreeSet<>(new BomDependencyComparator());
+        MavenXpp3Reader reader = new MavenXpp3Reader();
+        try {
+            Model model = reader.read(new FileReader(this.pomFileName));
+            DependencyManagement management = model.getDependencyManagement();
+            List<Dependency> externalBomDependencies = management.getDependencies().stream().filter(dependency -> dependency.getType().equals(POM_TYPE)).collect(Collectors.toList());
+            for (Dependency externalDependency : externalBomDependencies) {
+                externalDependencies.addAll(Utils.getPomFileContent(externalDependency));
+            }
+
+        } catch (XmlPullParserException e) {
+            e.printStackTrace();
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        } catch (IOException exception) {
+            exception.printStackTrace();
+        }
+
+        return externalDependencies;
+    }
+
+    private void rewriteBomFile() {
+        MavenXpp3Reader reader = new MavenXpp3Reader();
+        try {
+            Model model = reader.read(new FileReader(this.pomFileName));
+            DependencyManagement management = model.getDependencyManagement();
+            List<Dependency> dependencies = management.getDependencies();
+            dependencies.sort(new DependencyComparator());
+            management.setDependencies(dependencies);
+            MavenXpp3Writer writer = new MavenXpp3Writer();
+            writer.write(new FileWriter(this.pomFileName), model);
```

Comment:
Close all streams after reading/writing.

Reply:
This writer does not have a close.